src/lib/comms/sol-coap.c

```diff
@@ -662,8 +662,11 @@ pending_timeout_cb(void *data)
     struct pending_reply *reply = data;
     struct sol_coap_server *server = reply->server;
 
-    if (reply->cb((void *)reply->data, server, NULL, NULL))
-        return true;
+    if (reply->cb((void *)reply->data, server, NULL, NULL)) {
+        if (reply->freshness != PENDING_REPLY_FREE_ME) {
+            return true;
+        }
+    }
 
     sol_ptr_vector_remove(&server->pending, reply);
```

Comment:
I mean by ``sol_coap_cancel_send_packet()`` and friends, what you say about ``pending_reply_free()`` is true, but what about the also do ``sol_ptr_vector_del(&server->pending, i);``?